src/clib/pio_darray.c

```diff
@@ -361,34 +382,38 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
 
     /* Get memory for data. */
     if (arraylen > 0)
-        if (!(wmb->data = bgetr(wmb->data, (1+wmb->validvars)*arraylen*tsize)))
+        if (!(wmb->data = bgetr(wmb->data, (1 + wmb->validvars) * arraylen * tsize)))
             piomemerror(*ios, (1 + wmb->validvars) * arraylen * tsize, __FILE__, __LINE__);
 
     /* Get memory for ??? */
     if (!(wmb->vid = (int *)bgetr(wmb->vid, sizeof(int) * (1 + wmb->validvars))))
         piomemerror(*ios, (1 + wmb->validvars) * sizeof(int), __FILE__, __LINE__);
 
+    /* Get memory for ??? */
```

Comment:
wmb->frame is the record number, we assume that the variables in the wmb list may not all have the same unlimited dimension value although they usually do.